.github/workflows/build-wheels-upload-to-pypi.yml

```diff
@@ -0,0 +1,37 @@
+name: Build and Push to PyPI
+
+on:
+  release:
+    types: [published]
+
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, windows-2019, macOS-10.15]
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Used to host cibuildwheel
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel
+
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir wheelhouse
+
+      - name: Publish package to PyPI
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Comment:
Isn't this action only triggered when a release is published on GitHub (see lines 3-5)?

Reply:
Yup, this is probably redundant. This is the part is still a draft since for obvious reasons it was not run/tested yet. I know pypi has a test server where we could try it. I'm also considering turning this into a separate PR not to pack too many things into this one, any thoughts?